tmt/steps/prepare/install.py

```diff
@@ -163,8 +163,22 @@ def install_from_url(self) -> None:
     def install_debuginfo(self) -> None:
         """ Default base install method for debuginfo packages """
 
+    def make_cache(self) -> None:
+        """ Default base method to generate metadata cache"""
+
     def install(self) -> None:
         """ Perform the actual installation """
+        try:
+            self.perform_installation()
+        except Exception:
+            # Refresh cache in case of recent but not updated change to repodata
+            if str(self.guest.package_manager) in ["dnf", "dnf5", "yum"]:
```

Comment:
* I'd run `makecache()` in every case. We already know `apt` would benefit from this, Ubuntu and Debian images come without cache, `apt-get install tree` will fail without running `apt-get update` first. I believe we can find implementations for all supported package managers, even if it'd be a no-op.

Reply:
I've added all the other package managers.